base_layer/wallet/src/storage/sqlite_db.rs

```diff
@@ -243,6 +334,27 @@ impl WalletBackend for WalletSqliteDatabase {
                     Some(DbValue::ClientValue(v.value))
                 },
             },
+            DbKey::CommsAddress => self.get_comms_address(&conn)?,
+            DbKey::Identity => {
+                let secret_key = match self.get_comms_secret_key(&conn)? {
+                    Some(DbValue::CommsSecretKey(v)) => v,
+                    _ => return Ok(None),
+                };
+                let address = match self.get_comms_address(&conn)? {
```

Comment:
Why not rather return a valid `Option<CommsSecretKey>` here instead of wrapping it in a DbValue? Seems to add a lot of code in the checking stage

Reply:
This is part of this modular storage backend pattern that we conceived in the very beginning of the project to allow for multiple backends to be written that implement the `fetch(...)` and `write(...)` methods of the trait. But I agree I think the boilerplate this creates is not worth the benefit for a potential easier future integration of some other backend. Though removing it now is a cleanup refactor in itself so 🤷‍♂️

Reply:
Fair, but, I was more referring to the fact that `get_comms_address` could be written to return the result in the correct type (i.e. `CommsAddress`), and only the instance that needed to return a `DbValue` (i.e. `fetch`) wrap it.

Reply:
This is the same schematics we use all over the code base, from the BaseNode to the wallet for db write transactions.  I only did it this way to be consistent with the rest of the code.

Reply:
`fn get_comms_address` is a private function. 
You either have to unwrap the option here, and use above straight
OR
Use here straight, and wrap above

Reply:
I disagree that it is like that in the code base, and lines 330 and 324 are both examples of doing it correctly, and `get_comms_address` is a new method created in this PR. I would say change line 337 to 
```
DbKey::CommsAddress => self.get_comms_address(&conn)?.map(|v| DbValue::CommsAddress(v)),
```
and `get_comms_address` to:
```
fn get_comms_address(&self, conn: &SqliteConnection) -> Result<Option<CommsAddress>, WalletStorageError> 
```